src/hooks/useCachedTask/index.ts

```diff
@@ -0,0 +1,24 @@
+import { LogTypes } from "constants/enums";
+import { useLogContext } from "context/LogContext";
+import { LogkeeperTaskQuery, TaskQuery } from "gql/generated/types";
+import { cache } from "gql/GQLProvider";
+import { GET_LOGKEEPER_TASK, GET_TASK } from "gql/queries";
+
+export const useCachedTask = () => {
+  const { logMetadata } = useLogContext();
+  const { buildID, taskID, execution, logType } = logMetadata ?? {};
+
+  if (logType === LogTypes.RESMOKE_LOGS) {
+    const data = cache.readQuery<LogkeeperTaskQuery>({
+      query: GET_LOGKEEPER_TASK,
+      variables: { buildId: buildID },
+    });
+    return data?.logkeeperBuildMetadata?.task;
+  }
+
+  const data = cache.readQuery<TaskQuery>({
+    query: GET_TASK,
+    variables: { taskId: taskID, execution: Number(execution) },
+  });
```

Comment:
If you use useQuery you can probably also eliminate some of the complexity in the hook or the hook as a whole since the logic should be fairly straightforward.

Reply:
ohh my bad HAHA yeap I can update this

Reply:
Okay I tried it again, but this time I refactored some of our query logic for tasks. I'm thinking this should be okay given the default fetch policy of `cache-first`, so if it already exists in the cache we won't make a new network request (I checked the Network tab to verify this was true)